src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2848,4 +2968,205 @@ public function is_development_environment() {
+				}
+			}
+
+			$page_caching_response_headers[] = $response_headers;
+			$response_timing[]               = ( $end_time - $start_time ) * 1000;
+		}
+
+		return array(
+			'advanced_cache_present'        => (
+				file_exists( WP_CONTENT_DIR . '/advanced-cache.php' )
+				&&
+				( defined( 'WP_CACHE' ) && WP_CACHE )
+				&&
+				/** This filter is documented in wp-settings.php */
+				apply_filters( 'enable_loading_advanced_cache_dropin', true )
+			),
+			'page_caching_response_headers' => $page_caching_response_headers,
+			'response_timing'               => $response_timing,
+		);
+	}
+
+	/**
+	 * Get page cache details.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return WP_Error|array {
+	 *    Page cache detail or else a WP_Error if unable to determine.
+	 *
+	 *    @type string   $status                 Page cache status. Good, Recommended or Critical.
+	 *    @type bool     $advanced_cache_present Whether page cache plugin is available or not.
+	 *    @type string[] $headers                Client caching response headers detected.
+	 *    @type float    $response_time          Response time of site.
+	 * }
+	 */
+	public function get_page_cache_detail() {
+		$page_cache_detail = $this->check_for_page_caching();
+		if ( is_wp_error( $page_cache_detail ) ) {
+			return $page_cache_detail;
+		}
+
+		// Use the median server response time.
+		$response_timings = $page_cache_detail['response_timing'];
+		rsort( $response_timings );
+		$page_speed = $response_timings[ floor( count( $response_timings ) / 2 ) ];
+
+		// Obtain unique set of all client caching response headers.
+		$headers = array();
+		foreach ( $page_cache_detail['page_caching_response_headers'] as $page_caching_response_headers ) {
+			$headers = array_merge( $headers, array_keys( $page_caching_response_headers ) );
+		}
+		$headers = array_unique( $headers );
+
+		// Page caching is detected if there are response headers or a page caching plugin is present.
+		$has_page_caching = ( count( $headers ) > 0 || $page_cache_detail['advanced_cache_present'] );
+
+		if ( $page_speed && $page_speed < $this->get_good_response_time_threshold() ) {
+			$result = $has_page_caching ? 'good' : 'recommended';
+		} else {
+			$result = 'critical';
+		}
+
+		return array(
+			'status'                 => $result,
+			'advanced_cache_present' => $page_cache_detail['advanced_cache_present'],
+			'headers'                => $headers,
+			'response_time'          => $page_speed,
+		);
+	}
+
+	/**
+	 * Get the threshold below which a response time is considered good.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return int Threshold in milliseconds.
+	 */
+	public function get_good_response_time_threshold() {
+		/**
+		 * Filters the threshold below which a response time is considered good.
+		 *
+		 * @since 6.1.0
+		 * @param int $threshold Threshold in milliseconds. Default 600.
+		 */
+		return (int) apply_filters( 'page_cache_good_response_time_threshold', 600 );
```

Comment:
Maybe we should prefix this filter name with `site_status_`, similar to other Site Health check specific filters in WordPress core.